packages/remix-dev/vite/plugin.ts

```diff
@@ -816,16 +844,6 @@ export const remixVitePlugin: RemixVitePlugin = (remixUserConfig = {}) => {
 
         viteConfig = resolvedViteConfig;
 
-        ssrBuildContext =
-          viteConfig.build.ssr && viteCommand === "build"
-            ? {
-                isSsrBuild: true,
-                getBrowserManifest: createBrowserManifestForBuild,
-              }
-            : {
-                isSsrBuild: false,
-              };
-
```

Comment:
This logic moved earlier in the plugin lifecycle to the `config` hook, making use of `viteConfigEnv` rather than the using the resolved Vite config. This is safe because config resolution won't result in a different value for `ssr`.